src/wallet.ts

```diff
@@ -463,10 +463,8 @@ WalletMiddlewareOptions): JsonRpcMiddleware<any, Block> {
  * @param data - The data passed in typedSign request.
  */
 function validateVerifyingContract(data: string) {
-  const {
-    domain: { verifyingContract },
-  } = parseTypedMessage(data);
-  if (!isValidHexAddress(verifyingContract)) {
+  const { domain: { verifyingContract } = {} } = parseTypedMessage(data);
+  if (verifyingContract && !isValidHexAddress(verifyingContract)) {
```

Comment:
Does it matter if `verifyingContract` is an empty string?

Reply:
An empty value will not throw error.

Reply:
I think @jiexi meant to ask _should_ it throw an error.

I've just tested with v12.0.6 and it looks like an empty `verifyingContract` was accepted though, so continuing to accept it sounds reasonable for now. It did trigger a blockaid warning on v12.0.6, but no crash, and I was able to sign it.